### ISSUE TITLE
Pseudo-labels of self-training

### ISSUE THREAD

**Issue by joyolee** [facebookresearch/av_hubert#39]
Dear authors,

After reading your paper and looking into your public code, I would ask a question about the **pseudo-labels** of your self-training experiments. In Table 1 of your paper, the best result (26.9% WER) is achieved by **AV-HuBERT + Self-Training**. My understanding is that the pseudo-labels are the pseudo-labels of **VoxCeleb2**, not of LRS3, is it correct? Then the VoxCeleb2 with pseudo-labels and LRS3 with ground-truth labels are used for finetuning the AV-HuBERT to achieve 26.9% WER.

Thanks to your nicely organized code we can successfully run the finetuning experiments without self-training and reach a similar result. In order to reproduce the experiment with self-training, I would ask, if it is possible to share your pseudo-labels?

Thanks a lot in advance!

Best regards,
Zhengyang

**Comment by chevalierNoir** [comment]
Hi,

That's correct. For the 26.9% result, the pseudo-labels are **only** for VoxCeleb2. 

Regarding the pseudo-label files, unfortunately we are unable to release them for now due to some internal administrative issues. But the generation of pseudo-labels should be fairly straightforward with the models we release. You can run inference on VoxCeleb2 with an audio-visual speech recognizer fine-tuned with LRS3. The decoding command is [here](https://github.com/facebookresearch/av_hubert#seq2seq-decoding) and the hypothesis is in the json output. The list of VoxCeleb2 files we use is [here](https://github.com/facebookresearch/av_hubert/blob/main/avhubert/preparation/data/vox-en.id.gz) in the repo. Note in the paper we use the ASR model to generate the pseudo-labels thus the text generated above can differ. But given an audio-visual speech recognizer is of approximately same performance (sometimes better) as ASR, the result should be pretty close.
